commons/src/main/java/fr/free/nrw/commons/caching/CacheController.java

```diff
@@ -0,0 +1,94 @@
+package fr.free.nrw.commons.caching;
+
+import android.util.Log;
+
+import com.github.varunpant.quadtree.Point;
+import com.github.varunpant.quadtree.QuadTree;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import fr.free.nrw.commons.upload.MwVolleyApi;
+
+public class CacheController {
+
+    private double x, y;
+    private QuadTree quadTree;
+    private Point[] pointsFound;
+    private double xMinus, xPlus, yMinus, yPlus;
+
+    private static final String TAG = CacheController.class.getName();
+
+    public CacheController() {
+        quadTree = new QuadTree(-180, -90, +180, +90);
+    }
+
+    public void setQtPoint(double decLongitude, double decLatitude) {
+        x = decLongitude;
+        y = decLatitude;
+        Log.d(TAG, "New QuadTree created");
+        Log.d(TAG, "X (longitude) value: " + x + ", Y (latitude) value: " + y);
+    }
+
+    public void cacheCategory() {
```

Comment:
Hm.. I've been trying to re-write this, but the problem is that x and y are required for both findCategory() and cacheCategory(). In that case I thought it might make sense to set the x and y variables first, instead of setting them as parameters for both method calls?